benchmarking/run_benchmark.py

```diff
@@ -151,11 +171,319 @@ def stage_zero():
+                }
+            )
+            continue
+
+        print(
+            f"Succesfully benchmarked {dataset}. Errors: {old_err_count} --> {new_err_count}\n"
+        )
+        results.append(
+            {
+                "benchmark": dataset,
+                "initial_error_count": old_err_count,
+                "refactored_error_count": new_err_count,
+            }
+        )
+    print(f"Finished benchmarking datasets.")
+    print(f"Saving results to csv...")
+
+    stage_one_save_results(results)
+    return
+
+
+# Utility Functions
+def stage_one_annotate(dataset: str):
+    """
+    Runs NullAwayAnnotator on the passed dataset in order to prepare it for
+    refactoring
+    """
+
+    print(f"Annotating {dataset}...")
+
+    # Create config files
+    os.makedirs(ANNOTATOR_OUT_DIR, exist_ok=True)
+    with open(f"{ANNOTATOR_CONFIG}", "w+") as config_file:
+        _ = config_file.write(f"{NULLAWAY_CONFIG}/\t{SCANNER_CONFIG}\n")
+
+    # Clear annotator output folder (required for annotator to run)
+    shutil.rmtree(ANNOTATOR_OUT_DIR + "/0", ignore_errors=True)
+
+    build_cmd = " ".join(get_build_cmd(dataset))
+    cwd = os.getcwd()
+
+    annotate_cmd: list[str] = [
+        "java",
+        "-jar",
+        ANNOTATOR_JAR,
+        # Absolute path of an Empty Directory where all outputs of AnnotatorScanner and NullAway are serialized.
+        "-d",
+        ANNOTATOR_OUT_DIR,
+        # Command to run Nullaway on target; Should be executable from anywhere
+        "-bc",
+        f'"cd {cwd} && {build_cmd}"',
+        # Path to a TSV file containing value of config paths
+        "-cp",
+        ANNOTATOR_CONFIG,
+        # Fully qualified name of the @Initializer annotation.
+        "-i",
+        "com.uber.nullaway.annotations.Initializer",
+        # Checker name to be used for the analysis.
+        "-cn",
+        "NULLAWAY",
+        # Max depth to traverse as part of the analysis search
+        "--depth",
+        "10",
+    ]
+    res = subprocess.run(annotate_cmd, text=True, capture_output=True)
+    if res.returncode != 0:
+        print(
+            f"Annotation failed with exit code {res.returncode} for dataset {dataset}"
+        )
+        return
+
+    output_log_path = f"{OUTPUT_DIR}/{dataset}/annotator.txt"
+    with open(output_log_path, "w+") as f:
+        f.write(f"CMD:\n\t{" ".join(annotate_cmd)}\n")
+        f.write(f"STDOUT:\n\t{res.stdout}\n")
+        f.write(f"STDERR:\n\t{res.stderr}\n")
+
+    if DEBUG:
+        print(
+            f"Command used to annotate dataset {dataset}: \n\t{" ".join(annotate_cmd)}\n"
+        )
+
+    return
+
+
+def stage_one_refactor(dataset: str):
+    """
+    Runs VGR on the passed dataset
+    """
+
+    print(f"Refactoring {dataset}...")
+
+    output_file = f"{OUTPUT_DIR}/{dataset}/refactoring.txt"
+    dataset_path = f"{DATASETS_REFACTORED_DIR}/{dataset}"
+
+    refactor_cmd: list[str] = ["./gradlew", "run", f"'--args={dataset_path} All'"]
```

Comment:
I am concerned that because this applies all of the refactorings at once, the results might be confounded by refactorings that unintentionally _increase_ the error count, rather than decrease it. There's no reason to believe that our refactorings can only change the error count in one direction; instead, we're relying on the fact that any error that goes away after a semantics-preserving refactoring must have been a false positive. But, if we apply two refactorings - one of which removes an error, and the other of which unintentionally introduces a new error - we won't notice with this setup.